### PR TITLE
Fixed: Wrapping new dependencies used in File Splitting

### PR DESCRIPTION
**Pull request by iftwigs**

### CODE REVIEW

File: konfuzio_sdk/trainer/utils.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""Add utility common functions and classes to be used for AI Training."""
+
+from konfuzio_sdk.extras import torch, Trainer, TrainerCallback
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class LoggerCallback(TrainerCallback):
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/utils.py:10]: Why is this needed, can you add the explanation to the docstring?

File: tests/trainer/test_file_splitting.py [hunk 2]
```diff
@@ -291,73 +298,70 @@ def tearDownClass(cls) -> None:
...
-    """Test Multimodal File Splitting Model."""
+class TestTextualFileSplittingModel(unittest.TestCase):
+    """Test Textual File Splitting Model."""
 
     @classmethod
     def setUpClass(cls) -> None:
         """Initialize the tested class."""
-        cls.project = Project(id_=46)
-        cls.file_splitting_model = MultimodalFileSplittingModel(categories=cls.project.categories)
+        cls.project = Project(id_=14392)
```
Comment (zypriafl) [inline tests/trainer/test_file_splitting.py:313]: why do we change test project?

File: tests/trainer/test_file_splitting.py [hunk 3]
```diff
@@ -228,10 +235,10 @@ def test_splitting_ai_evaluate_full_on_training(self):
         """Test Splitting AI's evaluate_full on training Documents."""
         splitting_ai = SplittingAI(self.file_splitting_model)
         splitting_ai.evaluate_full(use_training_docs=True)
-        assert splitting_ai.full_evaluation.tp() == 3
-        assert splitting_ai.full_evaluation.fp() == 0
-        assert splitting_ai.full_evaluation.fn() == 0
-        assert splitting_ai.full_evaluation.tn() == 3
+        assert splitting_ai.full_evaluation.tp() >= 0
```
Comment (zypriafl) [inline tests/trainer/test_file_splitting.py:238]: Why do we remove the exact assertions?

File: konfuzio_sdk/trainer/file_splitting.py [hunk 4]
```diff
@@ -601,54 +889,53 @@ def __init__(self, model):
...
         :param document: The Document to predict the Pages of.
         :type document: Document
         :param inplace: Whether to predict the Pages on the original Document or on a copy.
         :type inplace: bool
+        :param split_on_blank_pages: A flag to enable splitting on blank Pages.
+        :type split_on_blank_pages: bool
         :returns: A list containing the modified Document.
         """
-        if inplace:
+        if inplace or self.model.requires_images:
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/file_splitting.py:906]: Why do we not deepcopy if images are required?

File: konfuzio_sdk/trainer/file_splitting.py [hunk 5]
```diff
@@ -205,9 +198,289 @@ def __init__(
...
+    def _image_transformation(self, page_images: List[PIL.Image.Image]) -> List[np.ndarray]:
+        """
+        Take an image and transform it into the format acceptable by the model's architecture.
+
+        :param page_images: A list of Pages' images to be transformed.
+        :type page_images: List[str]
+        :returns: A list of processed images.
+        """
+        images = []
+        for page_image in page_images:
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/file_splitting.py:244]: Is this the only place where we have this pre-processing?
Should it go to utils.py?

File: konfuzio_sdk/trainer/file_splitting.py [hunk 6]
```diff
@@ -205,9 +198,289 @@ def __init__(
...
+            image = page_image.convert('RGB')
+            image = image.resize((224, 224))
+            image = tf.keras.preprocessing.image.img_to_array(image)
+            image = image.reshape((1, image.shape[0], image.shape[1], image.shape[2]))
+            image = image[..., ::-1]  # replacement of keras's preprocess_input implementation because of dimensionality
+            image[0] -= 103.939
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/file_splitting.py:250]: Why do we do this?

File: konfuzio_sdk/trainer/file_splitting.py [hunk 7]
```diff
@@ -205,9 +198,289 @@ def __init__(
...
+        with torch.no_grad():
+            output = self.bert_model(**inputs)
+        txt_data = [output.pooler_output]
+        txt_data = [np.asarray(x).astype('float32') for x in txt_data]
+        txt_data = np.asarray(txt_data)
+        image = page.get_image().convert('RGB')
+        image = image.resize((224, 224))
+        image = tf.keras.preprocessing.image.img_to_array(image)
+        image = image.reshape((1, image.shape[0], image.shape[1], image.shape[2]))
+        image = image[..., ::-1]
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/file_splitting.py:375]: Maybe to be unified with above?

File: konfuzio_sdk/trainer/file_splitting.py [hunk 8]
```diff
@@ -205,9 +198,289 @@ def __init__(
...
+                        self.train_labels,
+                        epochs=epochs,
+                        verbose=1,
+                        batch_size=train_batch_size,
+                    )
+            else:
+                raise ValueError('Fitting on the GPU is impossible because there is no GPU available on the device.')
+        logger.info('Multimodal File Splitting Model fitting finished.')
+
+    def predict(self, page: Page, use_gpu: bool = False) -> Page:
```
Comment (zypriafl) [inline konfuzio_sdk/trainer/file_splitting.py:354]: I would add *args, **kwargs to not break old models, in case we add new arguments.

File: tests/trainer/test_file_splitting.py (hunk 3 above)
Comment (MohamedAmineDHIAB) [reply tests/trainer/test_file_splitting.py:239]: $f_1 = 1 \iff Precision = 1$ and $Recall = 1 \iff True$ $Negatives = Predicted$ $Negatives$ and $True$ $Positives = Predicted$ $Positives $

**Review by MohamedAmineDHIAB — APPROVED** [review]
